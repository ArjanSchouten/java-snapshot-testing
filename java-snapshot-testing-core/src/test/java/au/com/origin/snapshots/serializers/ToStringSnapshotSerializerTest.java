package au.com.origin.snapshots.serializers;

import lombok.AllArgsConstructor;
import lombok.Data;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class ToStringSnapshotSerializerTest {
    ToStringSnapshotSerializer serializer = new ToStringSnapshotSerializer();

    @Test
    void shouldSnapshotAnyString() {
        String result = serializer.apply(new Object[] { "John Doe" });
        assertThat(result).isEqualTo("[\nJohn Doe\n]");
    }

    @Test
    void shouldSnapshotUnicode() {
        String result = serializer.apply(new Object[] { "ðŸ¤”" });
        assertThat(result).isEqualTo("[\nðŸ¤”\n]");
    }

    @Test
    void shouldSnapshotAnyObject() {
        String result = serializer.apply(new Object[] { new Dummy(1, "John Doe") });
        assertThat(result).isEqualTo("[\nToStringSerializerTest.Dummy(id=1, name=John Doe)\n]");
    }

    @Test
    void shouldSnapshotMultipleObjects() {
        String result = serializer.apply(new Object[] { new Dummy(1, "John Doe"), new Dummy(2, "Sarah Doe") });
        assertThat(result).isEqualTo("[\nToStringSerializerTest.Dummy(id=1, name=John Doe)\nToStringSerializerTest.Dummy(id=2, name=Sarah Doe)\n]");
    }

    @Test
    void shouldSupportBase64SerializerType() {
        assertThat(serializer.getOutputFormat()).isEqualTo("TEXT");
    }

    @AllArgsConstructor
    @Data
    private static class Dummy {
        private int id;
        private String name;

        public String toString() {
            return "ToStringSerializerTest.Dummy(id=" + this.getId() + ", name=" + this.getName() + ")";
        }
    }

}
