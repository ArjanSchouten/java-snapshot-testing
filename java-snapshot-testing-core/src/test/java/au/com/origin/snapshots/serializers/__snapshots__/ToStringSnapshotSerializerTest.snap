au.com.origin.snapshots.serializers.ToStringSnapshotSerializerTest.canUseSnapshotConfigAnnotationAtClassLevel=[
ToStringSerializerTest.Dummy(id=1, name=John Doe)
]